### PR TITLE
1578 nakamoto remove anchor mode

### LINKED ISSUE #1578: [Nakamoto] Remove anchor-mode
... anywhere it does not have default values. (keep in some parts for subnets? or remove completely?)

### PR DESCRIPTION
**Pull request by janniks**
> This PR was published to npm with the version `6.13.0`
> e.g. `npm install @stacks/common@6.13.0 --save-exact`

- remove anchor mode from builder methods, default to `.Any` -- this byte will be ignored by nodes in >= Nakamoto

### CODE REVIEW

File: packages/stacking/src/utils.ts [hunk 1]
```diff
@@ -349,3 +359,135 @@ export function ensureLegacyBtcAddressForPox1({
...
+  if (/\.pox(-[2-3])?$/.test(contract)) {
+    // .pox, .pox-2 or .pox-3
+    if (signerKey || signerSignature || hasMaxAmount || hasAuthId) {
+      throw new Error(
+        'PoX-1, PoX-2 and PoX-3 do not accept a `signerKey`, `signerSignature`, `maxAmount` or `authId`'
+      );
+    }
+  } else {
+    // .pox-4 or later
+    if (!signerKey || !hasMaxAmount || typeof authId === 'undefined') {
```
Comment (MicaiahReid) [inline packages/stacking/src/utils.ts]:
```suggestion
    if (!signerKey || !hasMaxAmount || !hasAuthId) {
```
Is this logic right?

**Review by MicaiahReid — APPROVED** [review]
There's one small suggestion, but looks good to me!

File: packages/stacking/src/utils.ts (hunk 1 above)
Comment (janniks) [reply packages/stacking/src/utils.ts]: That is for `authId` being a number and `0` being falsy in javascript

**Comment by janniks** [comment]
* **#1654** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1654?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1685** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1685?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1684** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1684?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1683** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1683?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1682** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1682?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1657** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1657?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1656** <a href="https://app.graphite.dev/github/pr/hirosystems/stacks.js/1656?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `next`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @janniks and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Pull request merged**